### ISSUE TITLE
Nullable not being tracked through foreach of enumerable in jagged array

### ISSUE THREAD

**Issue by stephentoub** [dotnet/roslyn#34667]
**Version Used**:
3.0.0-beta4-19170-01+1deafee3682da88bf07d1c18521a99f47446cee8

**Repro**:
```C#
#nullable enable
using System.Collections.Generic;

class Program
{
    static void Main() { }

    static void Foo(IEnumerable<object[]?[]> source)
    {
        foreach (object[][] item in source) { }
    }
}
```
I'd expect this to warn about mismatched nullability in types, since the type of `item` should really be `object[]?[]`, but it compiles without warning.

This code does warn as expected, though:
```C#
object[][] first = source.GetEnumerator().Current;
```
citing:
```
Nullability of reference types in value of type 'object[]?[]' doesn't match target type 'object[][]'.
```

cc: @cston, @jaredpar, @danmosemsft, @dotnet/nullablefc

**Comment by gafter** [comment]
@stephentoub I'm guessing from this issue that you would expect an error on the code below?

``` c#
using System.Collections.Generic;

class Program
{
    static void Main() { }

    static void Foo(IEnumerable<object> source)
    {
        foreach (string s in source) { }
    }
}
```

**Comment by stephentoub** [comment]
> I'm guessing from this issue that you would expect an error on the code below?

I would, yes. Am I thinking about it incorrectly?

**Comment by 333fred** [comment]
This was fixed in https://github.com/dotnet/roslyn/pull/34926. Adding a test to verify.
